### PR TITLE
Update Apache Lucene to 9.7.0-snapshot-204acc3

### PR DESCRIPTION
**Pull request by reta**
### Description
Update Apache Lucene to 9.7.0-snapshot-204acc3,  bringing  https://github.com/apache/lucene/pull/12311 in

### Related Issues
N/A

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [X] Commits are signed per the DCO using --signoff
- [ ] Commit changes are listed out in CHANGELOG.md file (See: [Changelog](../blob/main/CONTRIBUTING.md#changelog))

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by reta** [comment]
@nknize @andrross mind taking a quick look folks? thank you

File: server/src/main/java/org/opensearch/index/search/MatchQuery.java [hunk 1]
```diff
@@ -56,6 +56,7 @@
 import org.apache.lucene.search.Query;
 import org.apache.lucene.search.TermQuery;
 import org.apache.lucene.util.QueryBuilder;
+import org.apache.lucene.util.TermAndBoost;
```
Comment (dblock) [inline server/src/main/java/org/opensearch/index/search/MatchQuery.java:59]: And where did this come from before? I see this from a quick search, does it need to be deleted?

```
private class TermAndBoost implements Comparable<TermAndBoost> {
```

**Review by dblock — CHANGES_REQUESTED** [review]

File: server/src/main/java/org/opensearch/index/search/MatchQuery.java (hunk 1 above)
Comment (reta) [reply server/src/main/java/org/opensearch/index/search/MatchQuery.java:59]: @dblock the `TermAndBoost` was an inner class of the QueryBuilder, but it was moved out to standalone class by https://github.com/apache/lucene/pull/12169

Comment (dblock) [reply server/src/main/java/org/opensearch/index/search/MatchQuery.java:59]: And https://github.com/opensearch-project/OpenSearch/blob/b17c88c8c33b9f627cbfbbd0cffba233bb17d6bb/server/src/main/java/org/apache/lucene/queries/BlendedTermQuery.java#L275 is not the same thing?

Comment (reta) [reply server/src/main/java/org/opensearch/index/search/MatchQuery.java:59]: Nope

**Review by dblock — APPROVED** [review]

File: server/src/main/java/org/opensearch/index/search/MatchQuery.java (hunk 1 above)
Comment (dblock) [reply server/src/main/java/org/opensearch/index/search/MatchQuery.java:59]: I'm surprised you didn't call me out on not understanding what's going on myself :)

**Pull request merged**